objects/oBoss1Head.object.gmx

```diff
@@ -85,7 +141,9 @@ sparkcounter = 0;
         <arguments>
           <argument>
             <kind>1</kind>
-            <string>expl = instance_create(x - 16 + random(32), y - 8 + random(24), oFXAnimSpark);
+            <string>///Burn Timer
+
+expl = instance_create(x - 16 + random(32), y - 8 + random(24), oFXAnimSpark);
```

Comment:
`random()` can usually be converted to `irandom()` for clarity in these cases, also should be turned into a var. I'll likely make a commit of my own tweaks later, just going over this for an initial review pass this evening.

Reply:
I specifically tried to avoid changing things like this and settled for mostly reorganizing the code for a more functional style and actually using switch statements. So most of the actual logic behind how the boss works is untouched, it just triggers everything more efficiently

Reply:
Cool, wasn't sure how far the scope of these refactors was. Miepee was sending me horror stories of the Tsumuri AI lol